node-data/src/events.rs

```diff
@@ -0,0 +1,37 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) DUSK NETWORK. All rights reserved.
+
+mod blocks;
+mod transactions;
+
+pub use blocks::BlockEvent;
+pub use transactions::TransactionEvent;
+
+#[derive(Clone, Debug)]
+pub struct NodeEvent {
+    pub target: String,
+    pub topic: String,
+    pub id: String,
+    pub data: String,
+}
```

Comment:
Few considerations:
1. Maybe those are `ChainEvent` more than `NodeEvent`…? Unless you're expecting to reuse the same structure for events that are not related to the chain (consensus, mempool, etc).

2. The RUES specs indicates the `target` as either this: `transactions` or this: `transactions:fe2ffdcdd27be82fc850bc269ff87fb8e72cf0cd4368e49fe947afeb48167f08`, calling the first part the `component` and the second part an `element` within the component. I would keep the consistency (it's not necessary an `id`) or, if you want to keep at "chain" level, and then have another structure on `Rusk` that is following the specs, then I would probably rename `target` in something else to avoid confusion (`category` maybe? Idk).
3. The `topic` and the `target` (component / category) should be `static str`, since the only part that vary is the `id`.
4. `data` could be an `Option` since you might want to emit an event with no `payload` – as I think we have for `on/transactions/removed` 
5. I assume you want to represent with this structure only emitted events, and you don't want to represent an event that the consumer wants to subscribe. Otherwise, also `id` would be `Option`; but personally this should be handled in the Rusk side, with the proper RUES Event struct, since it's common to both chain and vm.

Reply:
1. Since we are in the node_data module it should be ok to call them `NodeEvent`. However I agree with you that neither `node_data` is the right name, it should be called `chain_data`. So I decided to just call it `Event` and alias it as `ChainEvent` inside rusk
2. Changed to be aligned with Rues spec
3. Changed
4. Agree. Going to change to support multiple types of payload (as we do in Rues)
5. You assume right